app/controllers/files/files_controller.rb

```diff
@@ -85,4 +103,73 @@ def set_file
     def file_params
       params.require(:file).permit(:name, :foreign_ref, :filetype, :meta)
     end
+
+    def text_params
+      params.require(:text).permit(:content)
+    end
+
+
+  private
+    # Private: Create a text or url file containing content
+    #
+    # content   - the file content
+    #
+    # Returns a new BruseFile
+    def create_file(content)
+      if is_url?(content)
+        # return file name from url func
+        create_url_file(extract_url(content))
+      else
+        # return file name from text func
+        create_text_file content
+      end
+    end
+
+    # Private: Create a text file containing content
+    #
+    # content   - the file content
+    #
+    # Returns a new BruseFile
+    def create_text_file(content)
+      # generate file name
+      fileref = SecureRandom.uuid
```

Comment:
[`SecureRandom.uuid` doesn't accept any parameters!](http://ruby-doc.org/stdlib-2.2.2/libdoc/securerandom/rdoc/SecureRandom.html#method-c-uuid) It's 100% automagic.